src/Observers/MediaObserver.php

```diff
@@ -2,4 +2,241 @@
+
+                return false;
+            }
+
+            return true;
+        } catch (\Exception $e) {
+            $this->handleException($e, $model);
+
+            return false;
+        }
+    }
+
+    /**
+     * Sets geometry based on related model
+     *
+     * @return void
+     */
+    private function setGeometryFromModel(Media $media, Model $model)
+    {
+        try {
+            if ($model instanceof UgcPoi && $model->geometry) {
+                // For UgcPoi, directly copy the geometry
+                $media->geometry = $model->geometry;
+                Log::info("Media: geometry copied from UgcPoi {$model->id}");
+            } elseif ($model instanceof UgcTrack) {
+                $this->setGeometryFromTrack($media, $model);
+            } else {
+                // For other models, set default value
+                $this->setDefaultGeometry($media);
+            }
+        } catch (\Exception $e) {
+            $this->handleException($e, $media);
+            $this->setDefaultGeometry($media);
+        }
+    }
+
+    /**
+     * Sets geometry from a UgcTrack model
+     *
+     * @return void
+     */
+    private function setGeometryFromTrack(Media $media, UgcTrack $track)
+    {
+        try {
+            // For UgcTrack, calculate the centroid of the multilinestring
+            $centroid = DB::selectOne(
+                'SELECT ST_AsText(ST_Centroid(geometry)) as centroid FROM ugc_tracks WHERE id = ?',
+                [$track->id]
+            );
+
+            if ($centroid && $centroid->centroid) {
+                $media->geometry = $centroid->centroid;
+                Log::info("Media: geometry calculated from UgcTrack {$track->id}");
+            } else {
+                // If centroid calculation is not possible, set default value
+                $this->setDefaultGeometry($media);
+            }
+        } catch (\Exception $e) {
+            $this->handleException($e, $media);
+            $this->setDefaultGeometry($media);
+        }
+    }
+
+    /**
+     * Handles exceptions during media creation
+     *
+     * @return void
+     */
+    private function handleException(\Exception $e, Media $media)
+    {
+        Log::error('Error in MediaObserver-creating: '.$e->getMessage());
+        Log::error($e->getTraceAsString());
+        // In case of error, set default values to avoid crashes
+        $this->setDefaultValues($media);
+    }
+
+    /**
+     * Sets default values for app_id and geometry
+     *
+     * @return void
+     */
+    private function setDefaultValues(Media $media)
+    {
+        try {
+            // Set default app_id (1 is usually the main app ID)
+            if (! $media->app_id) {
+                $media->app_id = 1;
+            }
+
+            $this->setDefaultGeometry($media);
+        } catch (\Exception $e) {
+            Log::error('Error setting default values: '.$e->getMessage());
+            // Last resort fallback
+            $media->app_id = 1;
```

Comment:
l'app_id di default è quella legata all'utente corrente, auth()->user()->app_id, se l'utente non ce l'ha allora è giusto che dia errore, vogliamo che i media siano sempre legati ad un app_id ed una geometria, eventualmente li metteremo nullable in futuro